pentaho-js/src/main/javascript/cdf-legacy/components/Pentaho.Reporting.js

```diff
@@ -918,13 +918,20 @@ var SchedulePrptComponent = PrptComponent.extend({
       }
       parameters["jobParameters"] = jobParameters;
       var success = false;
+      var protectedUrl = wd.cdf.endpoints.getWebsocketWebapp() + "/api/scheduler/job";
```

Comment:
As you can see below, there exists a method whose soal purpose is abstract the scheduler job endpoint: `wd.cdf.endpoints.getScheduledJob()`.

The problem is that this method returns a URL which arlready includes `wd.cdf.endpoints.getWebapp()`, which is not absolute (hence we defined the new `getWebappFullUrl()` method).

It might not be a good idea to modify the existing `wd.cdf.endpoints.getScheduledJob()` method, as it could break existing code (ours or third-party). Additionally, all other endpoint methods seem to be returning URLs which are relative to the host.

Concluding, I propose you do:
1. Define a new method in `wd.cdf.endpoints` which returns the host (full) URL:
   ```js
    getHostFullUrl: function () { return document.location.protocol + "//" + document.location.host; },
   ```
2. Modify `getWebappFullUrl` to use that new method and remove redundancy.
3. Use the new method `getHostFullUrl()` and `getScheduledJob()` to build the value of `protectedUrl`.

Reply:
addressed review comments